### ISSUE TITLE
How can I add custom training on top of the pre-trained weights? 

### ISSUE THREAD

**Issue by eott42** [AlexeyAB/darknet#2927]
Hello, Alexey, I hope you can help me. It's probably simple, but I am new to all of this. Anyway, I am really having two problems. They are as follows:

1. I would like to use the pre-trained network (the one with 20 categories), but I would like to remove all categories that are unrelated to vehicles. 
2. I would like to train an additional category but I don't know how to add a new category in. I can only find tutorials to train from scratch. 

Regarding the **second issue**, there's not much more to say. I am unsure how to train in a new category on top of the categories that currently exist. If I train from scratch, my custom category works fine, but I want to include the training that already exists for the following categories: bicycle, bus, car, and motorbike.

Regarding the **first issue**, I followed [this](https://github.com/pjreddie/darknet/issues/142) thread, but the output is not correct (see image below).

voc_custom.names looks like: 
`bicycle`
`bus`
`car`
`motorbike`

voc_custom.data looks like:
`classes= 4`
`train  = data/train_voc.txt`
`valid  = data/2007_test.txt`
`#difficult = data/difficult_2007_test.txt`
`names = data/voc_custom.names`
`backup = backup/`

and yolov2-tiny-voc_custom.cfg, starting from the line "###########":

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=45
activation=linear
#45
[region]
anchors = 1.08,1.19,  3.42,4.41,  6.63,11.38,  9.42,5.11,  16.62,10.52
bias_match=1
classes=4
coords=4
num=5
softmax=1
jitter=.2
rescore=1

object_scale=5
noobject_scale=1
class_scale=1
coord_scale=1

absolute=1
thresh = .6
random=1

Then I run the line: `darknet.exe detector demo data/voc_custom.data cfg/yolov2-tiny-voc_custom.cfg yolov2-tiny-voc.weights test.mp4 -i 0` and this is what I get:

![help](https://user-images.githubusercontent.com/48610818/56113034-4b65cb80-5f22-11e9-99b7-090cf4e84b94.PNG)

**Comment by Hubert51** [comment]
Do you solve this problem?

**Comment by eott42** [comment]
@Hubert51 Sort of.  I don’t think Alexey is going to respond, so is there any reason to keep it open? I’m new to github so I don’t know what to do. I did figure out what to do though. I was going to delete the post but I can’t, so I just marked it closed.

**Comment by adriacabeza** [comment]
@eott42  you should post the solution you have found for your problem, maybe there is someone that will have the same issue and the only thing he will find is: 'I did figure out how to fix my problem'.

**Comment by haohlin** [comment]
> @eott42 you should post the solution you have found for your problem, maybe there is someone that will have the same issue and the only thing he will find is: 'I did figure out how to fix my problem'.

@eott42 Exactly me facing the same problem. Can the OP post the solution, please? 
I want to add new classes and meanwhile keep the original detection. How can I select the weights I need and load them to the network?

**Comment by eott42** [comment]
@haohlin Hey, I'll post the solution for you soon. I haven't done it yet because I'm a student right now and very busy.
